client/me/purchases/manage-purchase/notices.jsx

```diff
@@ -197,6 +234,137 @@ class PurchaseNotice extends Component {
 		);
 	}
 
+	renderOtherPurchasesExpiringNotice() {
+		const { translate, purchase, selectedSite, renewableSitePurchases } = this.props;
+
+		const showOtherPurchasesExpiringNotice =
+			selectedSite &&
+			! isAutoRenewing( purchase ) &&
+			renewableSitePurchases.length > 1 &&
+			renewableSitePurchases.some( ( otherPurchase ) => otherPurchase.id === purchase.id );
+
+		if ( ! showOtherPurchasesExpiringNotice ) {
+			return null;
+		}
+
+		return (
+			<>
+				<UpcomingRenewalsDialog
+					isVisible={ this.state.showUpcomingRenewalsDialog }
+					purchases={ renewableSitePurchases }
+					site={ selectedSite }
+					onConfirm={ this.handleExpiringNoticeRenewSelection }
+					onClose={ this.closeUpcomingRenewalsDialog }
+				/>
+				<Notice
+					className="manage-purchase__other-purchases-expiring-notice"
+					showDismiss={ false }
+					status="is-info"
+					icon="notice"
+					text={ this.getOtherPurchasesExpiringText() }
+				>
+					<NoticeAction onClick={ this.handleExpiringNoticeRenewAll }>
+						{ translate( 'Renew all' ) }
+					</NoticeAction>
+					{ this.trackImpression( 'other-purchases-expiring' ) }
+				</Notice>
+			</>
+		);
+	}
+
+	openUpcomingRenewalsDialog = () => {
+		this.trackClick( 'other-purchases-expiring-upcoming-renewals-dialog' );
+		this.setState( { showUpcomingRenewalsDialog: true } );
+	};
+
+	closeUpcomingRenewalsDialog = () => {
+		this.setState( { showUpcomingRenewalsDialog: false } );
+	};
+
+	getOtherPurchasesExpiringText() {
+		const { translate, purchase, moment, renewableSitePurchases } = this.props;
+		const expiry = moment( purchase.expiryDate );
+		const translateOptions = {
+			args: {
+				purchaseName: getName( purchase ),
+				expiry: expiry.fromNow(),
+			},
+			components: {
+				link: (
+					<button
+						className="manage-purchase__other-upgrades-button"
+						onClick={ this.openUpcomingRenewalsDialog }
+					/>
+				),
+			},
+		};
+
+		if ( isExpired( purchase ) ) {
+			if ( isDomainRegistration( purchase ) ) {
+				return translate(
+					'Your %(purchaseName)s domain expired %(expiry)s, and {{link}}other upgrades{{/link}} on this site will also be removed soon unless you take action.',
+					translateOptions
+				);
+			}
+
+			if ( isPlan( purchase ) ) {
+				return translate(
+					'Your %(purchaseName)s plan expired %(expiry)s, and {{link}}other upgrades{{/link}} on this site will also be removed soon unless you take action.',
+					translateOptions
+				);
+			}
+
+			return translate(
+				'Your %(purchaseName)s subscription expired %(expiry)s, and {{link}}other upgrades{{/link}} on this site will also be removed soon unless you take action.',
+				translateOptions
+			);
+		}
+
+		const hasPurchaseExpiringSoon = renewableSitePurchases.some(
+			( otherPurchase ) => moment( otherPurchase.expiryDate ).diff( Date.now(), 'days' ) < 30
```

Comment:
I think this should be limited to checking other purchases only (not the current purchase)?

Otherwise the displayed text says "and other upgrades on this site will also be removed soon" even when the other upgrades aren't the ones which will be removed soon.